assets/tests/components/ladderPage.test.tsx

```diff
@@ -525,6 +532,50 @@ describe("LadderPage", () => {
     )
     expect(result.queryByText("Vehicles")).toBeNull()
   })
+
+  test("opens the detour modal", async () => {
+    jest
+      .mocked(getTestGroups)
+      .mockReturnValue([
+        TestGroups.DetourRouteSelection,
+        TestGroups.DetoursPilot,
+        TestGroups.RouteLadderHeaderUpdate,
+      ])
+
+    const mockState = stateFactory.build({
+      routeTabs: [
+        routeTabFactory.build({
+          selectedRouteIds: ["1"],
+          isCurrentTab: true,
+        }),
+      ],
+    })
+    const { container } = render(
+      <StateDispatchProvider state={mockState} dispatch={jest.fn()}>
+        <BrowserRouter>
+          <RoutesProvider routes={routes}>
+            <LadderPage />
+          </RoutesProvider>
+        </BrowserRouter>
+      </StateDispatchProvider>
+    )
+
+    expect(container.querySelector(".c-new-route-ladder__header")).toBeVisible()
+
+    expect(
+      container.querySelector(".c-route-ladder__dropdown-button")
+    ).toBeVisible()
+
+    await userEvent.click(
+      container.querySelector(".c-route-ladder__dropdown-button")!
+    )
```

Comment:
nit(non-blocking): I don't like these selectors, so I opened #2657 to make it easier to test this button. Doesn't need to be addressed in this PR, but a followup PR to #2657 that adjust these tests would be nice.

Reply:
Oh yeah - that's a good call.